fdbclient/ManagementAPI.actor.cpp

```diff
@@ -3215,6 +3215,91 @@ ACTOR Future<size_t> getBulkDumpCompleteTaskCount(Database cx, KeyRange rangeToR
 	return completeTaskCount;
 }
 
+ACTOR Future<Optional<BulkDumpRestoreState>> getOngoingBulkDumpRestoreJob(Database cx) {
+	state RangeResult rangeResult;
+	state Transaction tr(cx);
+	loop {
+		try {
+			// At most one job at a time, so looking at the first returned range is sufficient
+			wait(store(rangeResult,
+			           krmGetRanges(&tr,
+			                        bulkDumpRestorePrefix,
+			                        normalKeys,
+			                        CLIENT_KNOBS->KRM_GET_RANGE_LIMIT,
+			                        CLIENT_KNOBS->KRM_GET_RANGE_LIMIT_BYTES)));
+			for (int i = 0; i < rangeResult.size() - 1; i++) {
+				if (rangeResult[i].value.empty()) {
+					continue;
+				}
+				BulkDumpRestoreState job = decodeBulkDumpRestoreState(rangeResult[i].value);
+				KeyRange jobRange = job.getRange();
+				ASSERT(!jobRange.empty() && jobRange.begin == rangeResult[i].key &&
+				       jobRange.end == rangeResult[i + 1].key);
+				return job;
+			}
+			return Optional<BulkDumpRestoreState>();
+		} catch (Error& e) {
+			wait(tr.onError(e));
+		}
+	}
+}
+
+ACTOR Future<bool> isBulkDumpRestoreAlive(Transaction* tr, Optional<UID> jobId = Optional<UID>()) {
+	state RangeResult rangeResult;
+	// At most one job at a time, so looking at the first returned range is sufficient
+	wait(store(rangeResult,
+	           krmGetRanges(tr,
+	                        bulkDumpRestorePrefix,
+	                        normalKeys,
+	                        CLIENT_KNOBS->KRM_GET_RANGE_LIMIT,
+	                        CLIENT_KNOBS->KRM_GET_RANGE_LIMIT_BYTES)));
+	for (int i = 0; i < rangeResult.size() - 1; i++) {
+		if (rangeResult[i].value.empty()) {
+			continue;
+		}
+		BulkDumpRestoreState job = decodeBulkDumpRestoreState(rangeResult[i].value);
+		KeyRange jobRange = job.getRange();
+		ASSERT(!jobRange.empty() && jobRange.begin == rangeResult[i].key && jobRange.end == rangeResult[i + 1].key);
+		if (jobId.present() && job.getJobId() != jobId.get()) {
+			continue;
+		}
+		return true;
+	}
+	return false;
+}
+
+ACTOR Future<Void> submitBulkDumpRestore(Database cx, BulkDumpRestoreState jobState) {
+	state Transaction tr(cx);
+	loop {
+		try {
+			bool existAny = wait(isBulkDumpRestoreAlive(&tr));
+			if (existAny) {
+				return Void(); // early exit
```

Comment:
We will return silently w/o submitting the new job? Shouldn't we let user know there is a job already running?